bloomfilter.c

```diff
@@ -1,21 +1,23 @@
 #include "bloomfilter.h"
+#include "bitarray.h"
+#include <stdbool.h>
+#include <stddef.h>
 
 // Each pair of i'th elements is a seed pair for a 'unique' hash function.
-int prime_seeds_1[k] = {1103, 761, 83};
-int prime_seeds_2[k] = {173, 587, 97};
+const static int prime_seeds_1[k] = {1103, 761, 83};
+const static int prime_seeds_2[k] = {173, 587, 97};
 
 /* Compute the modular hash of a char array.
  * @param a string (char array) to hash
  * @param R: a seed value, should be a prime
  * @param M: a seed value, should also be a prime
  */
-static int modular_hash(const char string[], int seed_1, int seed_2)
+int modular_hash(const char string[], int seed_1, int seed_2)
 {
     int hash = 0;
-    for (char character = *string; character != '\0'; character = *++string)
+    for (const char* character = string; *character != '\0'; character++)
```

Comment:
**suggestion (non-blocking):** Some questions for my understanding:

1. `const char string[]` an array of pointers to characters, and in line 18 we iterate through each character by incrementing the pointer until the value of `character` (the dereferenced pointer) is equal to the terminating '\0' ? 

2. Is `const char* character = string` initializing `character` as the first element of `string`?

3. `character++` Is this how we get the next char pointer?

Reply:
> **question (non-blocking):** Some questions for my understanding:
> 
>     1. `const char string[]` an array of pointers to characters, and in line 18 we iterate through each character by incrementing the pointer until the value of `character` (the dereferenced pointer) is equal to the terminating '\0' ?
> 
>     2. Is `const char* character = string` initializing `character` as the first element of `string`?
> 
>     3. `character++` Is this how we get the next char pointer?

const char[] (when passed as a function argument) is a pointer to an array of chars. 
const char* character=string takes a copy of the pointer that points to the array of chars and stores it as 'character'
*character returns what letter is stored at the address that character currently points to
character++ increments the pointer such that it points to the next character in the string

Reply:
Nice, thanks for the explanation!
It's cool that incrementing pointers can be done like that

Reply:
Ye under the hood they're just integers which is nice until you realise that x[3] is equivelant to 3[x] lol